assets/javascripts/admin_assets.js

```diff
@@ -121,37 +121,42 @@ function reloadAssetsTable() {
 }
 
 function deleteAsset(assetId) {
-  $.ajax({
-    url: urlWithBase('/api/v1/assets/' + assetId),
-    method: 'DELETE',
-    dataType: 'json',
-    success: function () {
+  fetchWithCSRF(urlWithBase(`/api/v1/assets/${assetId}`), {method: 'DELETE'})
+    .then(response => {
+      // not checking for status code as 404 case also returns proper json
+      return response.json();
+    })
+    .then(response => {
+      if (response.error) throw response.error;
       addFlash(
         'info',
-        'The asset was deleted successfully. The asset table\'s contents are cached. Hence the removal is not immediately visible. To update the view use the "Trigger asset cleanup" button. Note that this is an expensive operation which might take a while.'
+        "The asset was deleted successfully. The asset table's contents are cached." +
+          'Hence the removal is not immediately visible. To update the view use the "Trigger asset cleanup" button.' +
+          'Note that this is an expensive operation which might take a while.'
```

Comment:
```suggestion
        `The asset was deleted successfully. The asset table's contents are cached.
          Hence the removal is not immediately visible. To update the view use the "Trigger asset cleanup" button.
          Note that this is an expensive operation which might take a while.`
```

Reply:
this would actually change the contents of the string. Also no reason to use ` over ' unless expanding vars.